templates/amm.cpp.in

```diff
@@ -319,50 +320,59 @@ double calculate_amm_impl(const @ModelName@_mass_eigenstates& model, const softs
       return std::numeric_limits<double>::quiet_NaN();
    }
 
-   double val =
-      // vector form factor
-      1./2.*(form_factors[2] + form_factors[3]).real()
-      // reinstate the mass that was factored out in definition of form factor
-      * context.mass<Lepton>({@extraIdxUsageNoComma@})
-      // factor out e/(2*lepton pole mass)
-      / (unit_charge(context_base{model})/(2.*lepton_pole_mass<Lepton>(qedqcd@extraIdxUsage@)))
-      // definition of photon momentum flow for g-2 is opposite than for form factors
-      * (-1.);
+   double val = 0.;
+
+   // 1L
+   if (settings.get(Spectrum_generator_settings::calculate_amm) > 0.5) {
+      val +=
+         // vector form factor
+         1./2.*(form_factors[2] + form_factors[3]).real()
+         // reinstate the mass that was factored out in definition of form factor
+         * context.mass<Lepton>({@extraIdxUsageNoComma@})
+         // factor out e/(2*lepton pole mass)
+         / (unit_charge(context_base{model})/(2.*lepton_pole_mass<Lepton>(qedqcd@extraIdxUsage@)))
+         // definition of photon momentum flow for g-2 is opposite than for form factors
+         * (-1.);
+   }
 
-   // add 2-loop QED logarithms
-   val *= 1. + get_QED_2L<Lepton>(context, qedqcd, lepton_pole_mass<Lepton>(qedqcd@extraIdxUsage@));
+   // 2L QED
+   if (settings.get(Spectrum_generator_settings::calculate_amm) > 1.25) {
+      // add 2-loop QED logarithms
+      val *= 1. + get_QED_2L<Lepton>(context, qedqcd, lepton_pole_mass<Lepton>(qedqcd@extraIdxUsage@));
+   }
 
-   std::complex<double> valBarZee {0., 0.};
+   // Barr-Zee
+   if (settings.get(Spectrum_generator_settings::calculate_amm) > 1.75) {
+      std::complex<double> valBarZee {0., 0.};
 
 @AMMBarZeeCalculation@
-
-   if (!is_zero(valBarZee.imag())) {
-      ERROR("Error in the g-2 calculation! Form factor F2 should be real");
-      return std::numeric_limits<double>::quiet_NaN();
-   } else {
-      // @todo: enable once the calculation of Barr-Zee contributions is validated
-      // val += valBarZee.real();
+      if (!is_zero(valBarZee.imag())) {
```

Comment:
This check can be removed, if `valBarZee` is of type `double`.

Reply:
Done in bc37c3e37ba7f23a4df96e71871d94ec67ca2797.